Automattic/wp-calypso
# [ETK][GB 13.x] Fix loading of A8C-specific patterns in WPCOM

#### Changes proposed in this Pull Request

For full context, see: p1649463981925889-slack-C7YPUHBB2.

GB 13.0.0 loads block patterns asynchronously through a new endpoint introduced in [#](https://github.com/WordPress/gutenberg/pull/39185). This is not compatible with the `current_screen` hook for two reasons: execution order (which means we can't use that hook anymore) and the fact that even if we use another hook/filter, the data about the current screen is not available (though we are discussing ways to provide that data, [here](https://github.com/WordPress/gutenberg/issues/40305)). The workaround for now (for shipping GB 13.0.0 to WPCOM) is to just load all patterns regardless of the screen (site editor or regular editor).

#### Testing instructions

* Activate Gutenberg 13.0.0-rc.3 (or stable if available) in your WPCOM simple site by applying the `gutenberg-edge` sticker to it
* Sandbox `public-api.wordpress.com`
* Build ETK in this branch and sync to your sandbox using `yarn dev --sync`
* Now load the editor, bring up the block inspector and wait for the "Patterns" tab to render. Click it and you should see a relatively big list of categories. It may vary per theme, but it should be much more than 3, for example. See the video below for the ones that should load for TT2.
* Repeat the same for the Site Editor (make sure to use a theme that supports it, like TT2), You should see the same block pattern categories.
* cd to the ETK directory (`apps/editing-toolkit`) in Calypso and run tests: `yarn run test:php`. All tests should pass.


https://user-images.githubusercontent.com/81248/163303908-8730f1df-f725-469c-9d43-6b0c15b4f49b.mp4

### TODO

- [x] Test that the fix works in AT, make the necessary changes if needed
- [x] Ensure back compatibility with at least one gutenberg release
